18F/identity-idp
# LG-8860: Add reCAPTCHA fallback error screen

## 🎫 Ticket

[LG-8860](https://cm-jira.usa.gov/browse/LG-8860)

## 🛠 Summary of changes

Adds fallback error screen handling for reCAPTCHA failure scenario.

## 📜 Testing Plan

1. Generate reCAPTCHA credentials for use in your local environment: https://www.google.com/recaptcha/admin/create (Note you will need two sets of credentials, one for v2 and another for v3)
3. In `config/application.yml`:
   1. Assign credentials as `recaptcha_site_key_v2`, `recaptcha_site_key_v3`, `recaptcha_secret_key_v2`, and `recaptcha_secret_key_v3`
   3. Set `phone_setup_recaptcha_score_threshold` to `1.0`
4. Sign in
5. Add a phone
6. Submit a number (e.g. `5135550100` for U.S., `3065550100` for international) 
7. Observe error screen with reCAPTCHA checkbox if entering international number on Step 5, otherwise observe OTP confirmation screen
8. If presented with error screen with reCAPTCHA checkbox, confirm that clicking checkbox and successfully validating reCAPTCHA proceeds to OTP confirmation screen

## 👀 Screenshots

![Screen Shot 2023-02-14 at 10 48 43 AM](https://user-images.githubusercontent.com/1779930/218789626-5c36ab62-91e9-4df0-8ff3-8adf1ba2f5bf.png)
